WordPress/src/main/java/org/wordpress/android/WordPressDB.java

```diff
@@ -907,11 +907,29 @@ public Blog instantiateBlogByLocalId(int localId) {
      * self-hosted blogs that don't use jetpack)
      */
     public boolean isRemoteBlogIdDotComOrJetpack(int remoteBlogId) {
-        int localId = getLocalTableBlogIdForRemoteBlogId(remoteBlogId);
-        Blog blog = instantiateBlogByLocalId(localId);
+        Blog blog = instantiateBlogByRemoteId(remoteBlogId);
         return blog != null && (blog.isDotcomFlag() || blog.isJetpackPowered());
     }
 
+    public Blog instantiateBlogByRemoteId(int remoteBlogId) {
+        int localId = getLocalTableBlogIdForJetpackOrWpComRemoteSiteId(remoteBlogId);
+        Blog blog = instantiateBlogByLocalId(localId);
+        return blog;
+    }
+
+    public int getLocalTableBlogIdForJetpackOrWpComRemoteSiteId(int remoteBlogId) {
+        //first try checking if a jetpack site is configured on this device site list with that remote id
+        // workaround: There are 2 entries in the DB for each Jetpack blog linked with
+        // the current wpcom account. We need to load the correct localID here, otherwise options are
+        // blank
+        int localId = getLocalTableBlogIdForJetpackRemoteID(remoteBlogId, null);
+
+        if (localId == 0) {
+            localId = getLocalTableBlogIdForRemoteBlogId(remoteBlogId);
```

Comment:
You're right Danilo! I think I coded this solution trying to translate the name of the issue in a direct way. For reference here's the original issue name:

> Notifications: "Like" should not be an option for comments on Jetpack sites

So, what this code does is exactly that: only allow the "Like" option if the post is likeable, *and* this is not a Jetpack site.

That said, I'm following your suggestion and extend the solution for "Like" to be only an option for wp.com sites 👍